osquery/core/signing.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+#include <openssl/ecdsa.h>
+#include <openssl/pem.h>
+
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include <osquery/database.h>
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/tables/system/hash.h"
+
+#include "osquery/core/signing.h"
+
+namespace osquery {
+
+Status verifySignature(const std::string& b64Pub,
+                       const std::string& b64Sig,
+                       const std::string& message) {
+  Status ret;
+
+  std::string pub_key = base64Decode(b64Pub);
+  std::string sig = base64Decode(b64Sig);
+  // Erase key the header
+  pub_key.erase(0, 23);
+
+  EC_KEY* key;
+  key = EC_KEY_new_by_curve_name(NID_secp256k1);
+  const unsigned char* pub_bytes_ref = (unsigned char*)pub_key.c_str();
+  o2i_ECPublicKey(&key, &pub_bytes_ref, (long)pub_key.size());
+
+  // Check that we can load the public key
+  if (EC_KEY_check_key(key) == 1) {
```

Comment:
Nit, you can return hear here:
```
If (EC_KEY_check_key(key) != 1) {
  EC_KEY_free(key)
  return Status(1, "Unable to create public key");
}
```

The nitpick is, an early return allows for easier reading at the expense of freeing in two locations.